File: Leetcode/leetcode-1436-DestinationCity-[array]-[easy]/Solution.py
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        map = dict({})
        for pair in paths:
            start, destination = pair
            map.update({ start: destination })
        
        finalDestination = ""
        for destination in map.values():
            if (map.get(destination) == None):
                finalDestination = destination

        return finalDestination

File: Leetcode/leetcode-1436-DestinationCity-[array]-[easy]/Solution.java
class Solution {
    public String destCity(List<List<String>> paths) {
        HashMap<String, String> map = new HashMap<>();
        for (List<String> pair : paths) {
            map.put(pair.get(0), pair.get(1));
        }

        String destination = "";
        for (String possibleDestination : map.values()) {
            if (map.get(possibleDestination) == null) {
                destination = possibleDestination;
            }
        }

        return destination;
    }
}